File: src/person_director.cpp
#include "include/person_director.h"

person *person_director::create_person(person_builder &pb) {
  if (pb.check_args()) throw std::string("Bad Input");

  pb.create_person();

  pb.set_person_first_name();
  pb.set_person_second_name();
  pb.set_person_address();
  pb.set_person_passport_id();
  pb.get_person()->assign_id();
  pb.get_person()->update();

  return pb.get_person();
}

File: MIPT_bank_app1/MIPT_bank_app/app/src/main/java/com/example/mipt_bank_app/PersonDirector.java
package com.example.mipt_bank_app;

public class PersonDirector {
    Person createPerson(PersonBuilder pb) {
        try {
            if (pb.checkArgs()) throw new ArithmeticException("Bad Input");
        } catch (ArithmeticException e) {
            System.out.println(e);
        }
        pb.createPerson();
        pb.setPersonFirstName();
        pb.setPersonSecondName();
        pb.setPersonAddress();
        pb.setPersonPassportId();
        pb.getPerson().assignId();
        pb.getPerson().update();
        return pb.getPerson();
    }

}
